src/manifest/manifest_test.ts

```diff
@@ -972,6 +972,49 @@ Deno.test("SlackManifest() oauth2 providers get set properly", () => {
   });
 });
 
+Deno.test("SlackManifest() oauth2 providers get set properly with use_pkce", () => {
+  const providerKey1 = "test_provider_with_with_pkce_true";
+  const providerKey2 = "test_provider_with_with_pkce_false";
+
+  const Provider1 = DefineOAuth2Provider({
+    provider_key: providerKey1,
+    provider_type: Schema.providers.oauth2.CUSTOM,
+    options: {
+      "client_id": "123.456",
+      "scope": ["scope_a", "scope_b"],
+      "use_pkce": true,
+    },
+  });
+
+  const Provider2 = DefineOAuth2Provider({
+    provider_key: providerKey2,
+    provider_type: Schema.providers.oauth2.CUSTOM,
+    options: {
+      "client_id": "123.456",
+      "scope": ["scope_a", "scope_b"],
+      "use_pkce": false,
+    },
+  });
+
+  const definition: SlackManifestType = {
+    name: "Name",
+    description: "Description",
+    icon: "icon.png",
+    botScopes: [],
+    externalAuthProviders: [Provider1, Provider2],
+  };
+  assertEquals(definition.externalAuthProviders, [Provider1, Provider2]);
+  const Manifest = new SlackManifest(definition);
+  const exportedManifest = Manifest.export();
+
+  assertEquals(exportedManifest.external_auth_providers, {
+    "oauth2": {
+      "test_provider_with_with_pkce_true": Provider1.export(),
```

Comment:
I think the only thing this test tests is that the provider key is set properly on the exported manifest. Other than that assertion, this test will always pass, as the manifest values for any oauth provider will always be the output of the provider's `export` method. For example, in the future, if I change the [`Provider.export` method](https://github.com/slackapi/deno-slack-sdk/blob/main/src/providers/oauth2/mod.ts#L28) to `delete` the `pkce` field, this test will still pass.

I think we want to change this test to ensure that the `options.use_pkce` field is set properly on the manifest in both cases: it is `true` in one case and `false` in the other.

Reply:
I tried with some test mutation and seems like I can make the test fail like this:
<img width="866" alt="Screenshot 2023-08-23 at 1 35 15 PM" src="https://github.com/slackapi/deno-slack-sdk/assets/98924772/dea90c4e-3adc-4649-a1d6-ec2f8ee3dace">

Reply:
Do you think this would work? ☝️

Reply:
Right, that fails because the key of Provider2 is being checked against the export of Provider1, so we are checking that two different things are equal.

Reply:
I'd suggest enumerating specifically the fields that you expect in each case. Something like:

```
assertEquals(exportedManifest.external_auth_providers.oauth2.test_provider_with_with_pkce_true.options.use_pkce, true);
assertEquals(exportedManifest.external_auth_providers.oauth2.test_provider_with_with_pkce_false.options.use_pkce, false);
```

Reply:
Thanks for the suggestion! Added the changes ! 🙇‍♀️

Reply:
@filmaj  Please do not merge it until the web app changes are merged! Thanks!